Mathlib/AlgebraicGeometry/GammaSpecAdjunction.lean

```diff
@@ -544,6 +544,12 @@ variable (φ) in
 @[simp] lemma Spec.preimage_map : Spec.preimage (Spec.map φ) = φ :=
   Spec.map_injective (Spec.map_preimage (Spec.map φ))
 
+/-- Useful for replacing `f` by `Spec.map φ` everywhere in proofs. -/
+lemma Spec.exists_preimage_map {R S : CommRingCat} (f : Spec S ⟶ Spec R) :
+    ∃ φ, f = Spec.map φ := by
+  use Spec.preimage f
+  simp
```

Comment:
```suggestion
lemma Spec.map_surjective {R S : CommRingCat} :
    Function.Surjective (Spec.map : (R ⟶ S) → _) := by
  intro f
  use Spec.preimage f
  simp
```
I think this is defeq but has the potential to unlock dot notations for some lemmas.